front/src/reducers/globalMap/index.ts

```diff
@@ -0,0 +1,83 @@
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useOsrdContext } from 'common/osrdContext';
+import type { RootState } from 'reducers';
+
+import type { MapSettings } from './types';
+
+export const defaultMapSettings: MapSettings = {
+  mapStyle: 'normal',
+  showIGNBDORTHO: false,
+  showIGNSCAN25: false,
+  showIGNCadastre: false,
+  showOSM: true,
+  showOSM3dBuildings: false,
+  showOSMtracksections: false,
+  terrain3DExaggeration: 0,
+  smoothTravel: false,
+  layersSettings: {
+    buffer_stops: false,
+    electrifications: false,
+    neutral_sections: false,
+    detectors: false,
+    operational_points: true,
+    routes: false,
+    signals: false,
+    sncf_psl: false,
+    speed_limits: false,
+    speedlimittag: null,
+    switches: false,
+    tvds: false,
+    platforms: false,
+  },
+  viewport: {
+    latitude: 48.32,
+    longitude: 2.44,
+    zoom: 6.2,
+    bearing: 0,
+    pitch: 0,
+    padding: { top: 0, left: 0, bottom: 0, right: 0 },
+    width: 0,
+    height: 0,
+  },
+  mapSearchMarker: undefined,
+  lineSearchCode: undefined,
+};
+
+export function buildMapStateReducer<T extends { mapSettings: MapSettings }>() {
+  return {
+    updateMapSettings: (state: T, action: PayloadAction<Partial<MapSettings>>) => {
+      state.mapSettings = {
+        ...state.mapSettings,
+        ...action.payload,
+      };
+    },
+  };
+}
+
+export const useMapSettings = (): MapSettings => {
+  const { selectors } = useOsrdContext();
+
+  if (!selectors || !('getMapSettings' in selectors)) {
```

Comment:
selectors is always defined
```suggestion
  if (!('getMapSettings' in selectors)) {
```